SPECIFICATION.md

```diff
@@ -118,46 +117,50 @@ Each sprite is a collection of cells or "sprite parts", small textures taken fro
+| u8     | (Padding)      | 8-bit padding                                                  |
 
 ### Animation:
 
-Each animation is a collection of animation frames, which are references to sprites with specific
+Each animation is a collection of animation keys, which are references to sprites with specific
 added properties
 
 | Type             | Name             | Description      |
 | ---------------- | ---------------- | ---------------- |
-| u32              | Number of frames | Number of frames |
-| 20 bytes / frame | Animation frames | List of frames   |
-
-Each frame has the following structure:
-
-| Type    | Name      | Description                                                                        |
-| ------- | --------- | ---------------------------------------------------------------------------------- |
-| u16     | Sprite    | Number of the sprite that this animation frame uses                                |
-| u16     | Duration  | Amount of time that the frame will last. Unit is variable, sometimes BPM-dependent |
-| f32     | X scaling | Multiplier to the frame's X scale (width)                                          |
-| f32     | Y scaling | Multiplier to the frame's Y scale (height)                                         |
-| f32     | Rotation  | Rotation of the frame (in degrees?)                                                |
-| u8      | Opacity   | Opacity of the frame                                                               |
-| 3 bytes | Unknown   | Padding?                                                                           |
\ No newline at end of file
+| u16              | Number of keys   | Number of keys   |
+| u16              | (Padding)        | 16-bit padding   |
+| 24 bytes / key   | Animation keys   | List of keys     |
+
+Each key has the following structure:
+
+| Type   | Name           | Description                                                           |
+| ------ | -------------- | --------------------------------------------------------------------- |
+| u16    | Sprite index   | Index of the sprite shown on this key                                 |
+| u16    | Duration       | Amount of frames that this key will be shown for (FPS is not fixed)   |
+| s16    | Position X     | Additive X offset for the sprite                                      |
+| s16    | Position Y     | Additive Y offset for the sprite                                      |
+| f32    | Scale X        | The sprite's X scaling factor                                         |
+| f32    | Scale Y        | The sprite's Y scaling factor                                         |
+| f32    | Angle          | Additive rotation angle for the sprite (in degrees)                   |
+| u8     | Opacity        | Opacity of the sprite                                                 |
+| u24    | (Padding)      | 24-bit padding                                                        |
```

Comment:
personally i'd rather write a 24-bit padding as u8[3] or, as I put there, "3 bytes", since u24 is not a primitive type in most programming languages

Reply:
Good point